sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegen.scala

```diff
@@ -114,27 +117,67 @@ trait CodegenSupport extends SparkPlan {
   }
 
   /**
-    * Consume the columns generated from it's child, call doConsume() or emit the rows.
+    * Returns source code to evaluate all the variables, and clear the code of them, to prevent
+    * them to be evaluated twice.
+    */
+  protected def evaluateVariables(variables: Seq[ExprCode]): String = {
```

Comment:
Can you update the comment for ExprCode.code to specify what it means when it is empty.